packages/portfolio-contract/src/pos-gmp.flows.ts

```diff
@@ -49,60 +48,115 @@ import {
 } from './portfolio.flows.ts';
 import { TxType } from './resolver/constants.js';
 import type { ResolverKit } from './resolver/resolver.exo.ts';
+import type { TxId } from './resolver/types.ts';
 import type { PoolKey } from './type-guards.ts';
+import { predictWalletAddress } from './utils/evm-orch-factory.ts';
 
 const trace = makeTracer('GMPF');
 const { keys } = Object;
 
-export const provideEVMAccount = async (
+export type GMPAccountStatus = GMPAccountInfo & {
+  /** created and ready to accept GMP messages */
+  ready: Promise<unknown>;
+};
+
+export const provideEVMAccount = (
   chainName: AxelarChain,
+  chainInfo: BaseChainInfo,
   gmp: {
     chain: Chain<{ chainId: string }>;
     fee: NatValue;
-    axelarIds: AxelarId;
-    evmGas: bigint;
   },
   lca: LocalAccount,
   ctx: PortfolioInstanceContext,
   pk: GuestInterface<PortfolioKit>,
-) => {
-  await null;
-  const found = pk.manager.reserveAccount(chainName);
-  if (found) {
-    return found as unknown as Promise<GMPAccountInfo>; // XXX Guest/Host #9822
-  }
-
-  // We have the map entry reserved - use critical section pattern
-  try {
-    const pId = pk.reader.getPortfolioId();
-    const traceChain = trace.sub(`portfolio${pId}`).sub(chainName);
-    const axelarId = gmp.axelarIds[chainName];
-    const target = {
-      axelarId,
-      remoteAddress: ctx.contracts[chainName].factory,
+): GMPAccountStatus => {
+  const pId = pk.reader.getPortfolioId();
+  const traceChain = trace.sub(`portfolio${pId}`).sub(chainName);
+
+  const predictAddress = (owner: Bech32Address) => {
+    const contracts = ctx.contracts[chainName];
+    const remoteAddress = predictWalletAddress({
+      owner,
+      factoryAddress: contracts.factory,
+      gasServiceAddress: contracts.gasService,
+      gatewayAddress: contracts.gateway,
+      // XXX converting a 9k hex string to bytes for every account is a waste.
+      // But Uint8Array is not Passable. Pass it via prepareXYZ()?
+      walletBytecode: hexToBytes(ctx.walletBytecode.replace(/^0x/, '')),
```

Comment:
this should not need a regexp! and it should not be agnostic about whether the string starts with '0x'. That should be consistent and addressed statically